doc/source/ray-air/doc_code/air_ingest_migration.py

```diff
@@ -0,0 +1,62 @@
+# flake8: noqa
+# isort: skip_file
+
+# __legacy_api__
+import random
+import ray
+
+from ray.air.config import ScalingConfig, DatasetConfig
+from ray.data.preprocessors.batch_mapper import BatchMapper
+from ray.train.torch import TorchTrainer
+
+train_ds = ray.data.range_tensor(1000)
+test_ds = ray.data.range_tensor(10)
+
+# A randomized preprocessor that adds a random float to all values.
+add_noise = BatchMapper(lambda df: df + random.random(), batch_format="pandas")
+
+my_trainer = TorchTrainer(
+    lambda: None,
+    scaling_config=ScalingConfig(num_workers=1),
+    datasets={
+        "train": train_ds,
+        "test": test_ds,
+    },
+    dataset_config={
+        "train": DatasetConfig(
+            split=True,
+            # Apply the preprocessor for each epoch.
+            per_epoch_preprocessor=add_noise,
+        ),
+        "test": DatasetConfig(
+            split=False,
+        ),
+    },
+)
+my_trainer.fit()
+# __legacy_api_end__
+
+# __new_api__
+from ray.train.data_config import DataConfig
```

Comment:
We should be clear here that `ray.train.data_config` is not a public API. All the configs should be included via `ray.train` directly going forward. Also see the PR description of https://github.com/ray-project/ray/pull/36706 and the REP https://github.com/ray-project/enhancements/pull/36 :)

Reply:
@ericl maybe we should rename `data_config` to `_data_config` so there are not two ways to do this and people get confused?

Reply:
We should put it in a _internal folder then probably, to follow convention of other libraries.

Reply:
Ok, let's replace `ray.train.data_config` with `ray.train` everywhere in this PR before merging, and then we can do the `_internal` refactor later as time permits (doesn't need to be part of 2.6).

Reply:
So I assume nothing is needed in this PR, right?

Reply:
It is -- you need to replace `ray.train.data_config.DataConfig` with `ray.train.DataConfig` everywhere :)

Reply:
oh, I didn't know it's already in `ray.train`. I think there may be other places using `ray.train.data_config` as well. Will update all.

Reply:
Thanks, that's awesome :)